Library/Homebrew/resource.rb

```diff
@@ -53,10 +53,12 @@ def owner=(owner)
     @owner = owner
     patches.each { |p| p.owner = owner }
 
-    return if !owner.respond_to?(:full_name) || owner.full_name != "ca-certificates"
+    return if !(owner.respond_to?(:full_name) && owner.full_name == "ca-certificates") &&
+              !(!DevelopmentTools.curl_handles_most_https_certificates? && !HOMEBREW_BREWED_CURL_PATH.exist?)
     return if Homebrew::EnvConfig.no_insecure_redirect?
 
-    @insecure = !specs[:bottle] && !DevelopmentTools.ca_file_handles_most_https_certificates?
+    @insecure = !specs[:bottle] && (!DevelopmentTools.ca_file_handles_most_https_certificates? ||
+                                    !DevelopmentTools.curl_handles_most_https_certificates?)
```

Comment:
Similarly here: some gnarly logic that would read more easily with intermediate variables or functions.

Reply:
The first block is now unchanged since `--insecure` is again being passed only for `ca-certificates` installation; for the second I was able to use the functions I just added.